enterprise/cmd/frontend/userlimitchecker/limitchecker.go

```diff
@@ -0,0 +1,102 @@
+package userlimitchecker
+
+import (
+	"context"
+	"time"
+
+	ps "github.com/sourcegraph/sourcegraph/enterprise/cmd/frontend/internal/dotcom/productsubscription"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/txemail"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+// send email to site admins if approaching user limit on active license
+func sendApproachingUserLimitAlert(ctx context.Context, db database.DB) error {
+	userCount, err := getUserCount(ctx, db)
+	if err != nil {
+		return errors.Wrap(err, "could not get user count")
+	}
+
+	userLimit, err := getLicenseUserLimit(ctx, db)
+	if err != nil {
+		return errors.Wrap(err, "could not get license user limit")
+	}
+
+	siteAdminEmails, err := getSiteAdminEmails(ctx, db)
+	if err != nil {
+		return errors.Wrap(err, "could not get site admins")
+	}
+
+	replyTo := "support@sourcegraph.com"
+	messageId := "approaching_user_limit"
+
+	if approachingOrOverUserLimit(userCount, userLimit) {
+		if err := txemail.Send(ctx, "approaching_user_limit", txemail.Message{
+			To:        siteAdminEmails,
+			Template:  approachingUserLimitEmailTemplate,
+			MessageID: &messageId,
+			ReplyTo:   &replyTo,
+			Data: struct {
+				RemainingUsers int
+			}{
+				RemainingUsers: userLimit - userCount,
+			},
+		}); err != nil {
+			return errors.Wrap(err, "could not send email")
+		}
+	}
+
+	return nil
+}
+
+func approachingOrOverUserLimit(userCount, userLimit int) bool {
+	return userCount >= userLimit-5
+}
+
+func getUserCount(ctx context.Context, db database.DB) (int, error) {
+	userStore := db.Users()
+	userCount, err := userStore.Count(ctx, &database.UsersListOptions{})
+	if err != nil {
+		return 0, err
+	}
+	return userCount, nil
+}
+
+func getLicenseUserLimit(ctx context.Context, db database.DB) (int, error) {
+	items, err := ps.NewDbLicense(db).List(ctx, ps.DbLicencesListNoOpt())
+	if err != nil {
+		return 0, err
+	}
+
+	for _, item := range items {
+		if item.LicenseExpiresAt.UTC().After(time.Now().UTC()) {
+			return *item.LicenseUserCount, nil
+		}
+	}
+
+	return 0, nil
+}
+
+func getSiteAdminEmails(ctx context.Context, db database.DB) ([]string, error) {
+	var siteAdminEmails []string
+	users, err := db.Users().List(ctx, &database.UsersListOptions{})
+	if err != nil {
+		return siteAdminEmails, err
```

Comment:
```suggestion
		return nil, err
```
Returning `nil` along with an error is a more clear signal of an error condition than returning a non-`nil` value.